onnxscript/ir/_linked_list.py

```diff
@@ -50,12 +42,10 @@ def __init__(self, owner: DoublyLinkedList[TLinkable], value: TLinkable | None)
                 the link box is considered erased (default). The root box of the list
```

Comment:
Seems to prevent us from using a container where None is a legal value, preventing its general applicability in other scenarios. Not a major concern right now, but I have a vague feeling that's not a good idea.

Reply:
I guess it is primarily an optimization right now to avoid a separate `erased` field.

Reply:
Right I should have called that out. 

>  preventing its general applicability in other scenarios

This is valid. Maybe retaining the erased field is a good idea?

Reply:
Now raising an error if value is None.